tests/test_interaction_restrictions.py

```diff
@@ -67,42 +87,55 @@ def test_metrics_to_json():
     interaction_restrictions.parse_interaction_restrictions(
         PDB_FILE, yaml_obj.interaction_restrictions
     )
+
+    errors = []
     # Check metrics json output
-    expected_output = open(EXPECTED_METRICS, "r")
-    assert interaction_restrictions.metrics_to_json() == expected_output.read()
+    metrics = interaction_restrictions.metrics_to_json()
+    expected_metrics_output = open(EXPECTED_METRICS, "r").read()
+    if not metrics == expected_metrics_output:
+        errors.append(f"Error in metrics json: {metrics} == {expected_metrics_output} ")
+
+    # Check conditions json output
+    conditions = interaction_restrictions.conditions_to_json()
+    expected_conditions_output = open(EXPECTED_CONDITIONS, "r").read()
+    if not conditions == expected_conditions_output:
+        errors.append(
+            f"conditions json assert: {conditions} == {expected_conditions_output} "
+        )
 
+    # assert no error message has been registered, else print messages
+    assert not errors
 
-def test_conditions_to_json():
-    # Parse yaml file
-    yaml_obj = yp.YamlParser(ARGS_1, vf.VALID_FLAGS_PLATFORM)
-    yaml_obj.read()
-    # Build interaction restrictions
-    interaction_restrictions = ir.InteractionRestrictionsBuilder()
-    interaction_restrictions.parse_interaction_restrictions(
-        PDB_FILE, yaml_obj.interaction_restrictions
-    )
-    # Check metrics json output
-    expected_output = open(EXPECTED_CONDITIONS, "r")
-    assert interaction_restrictions.conditions_to_json() == expected_output.read()
+
+def test_SyntaxError_exception_1():
+    """
+    Unit test to throw an exception in a syntax error in input file (wrong number
+    of atoms).
+    """
+    check_SyntaxError_exception(ARGS_2)
 
 
-def test_SyntaxError_exception():
+def test_SyntaxError_exception_2():
+    """
+    Unit test to throw an exception in a syntax error in input file (poorly defined metric).
+    """
+    check_SyntaxError_exception(ARGS_3)
+
+
```

Comment:
You can run the same test with different inputs using pytest parametrization, all you need to do is add a decorator, e.g. `@pytest.mark.parametrize("file", [ARGS_2, ARGS_3])`

https://docs.pytest.org/en/stable/parametrize.html